vulnerabilities/importers/pysec.py

```diff
@@ -0,0 +1,178 @@
+        severity.append(
+            VulnerabilitySeverity(
+                system=SCORING_SYSTEMS["generic_textual"],
+                value=raw_data["ecosystem_specific"]["severity"],
+            )
+        )
+    else:
+        severity = []
+        logger.warning(f"severity not found- {raw_data['id']!r}")
+
+    references = []
+    if "references" in raw_data:
+        references = [
+            Reference(url=ref["url"], severities=severity) for ref in raw_data["references"] if ref
+        ]
+
+    affected_package = []
+    if "affected" in raw_data:
+        for affected_pkg in raw_data["affected"]:
+            package = affected_pkg["package"]
+            if "ecosystem" in package and "name" in package:
+                purl = PackageURL(type=package["ecosystem"], name=package["name"])
+            else:
+                purl = ""
+                logger.error(f"purl affected_pkg not found - {raw_data['id'] !r}")
+
+            affected_version_range = None
+            if "versions" in affected_pkg:
+                affected_version_range = (
+                    # HINT:  Values larger than 1/3 of a buffer page cannot be indexed.
+                    PypiVersionRange(affected_pkg["versions"][slice(200)])
+                )
+
+            if "ranges" in affected_pkg:
+                for fix_range in affected_pkg["ranges"]:
+                    fixed_version = set()
+                    if "type" in fix_range:
+                        if fix_range["type"] == "ECOSYSTEM":
+                            filter_fixed = list(
+                                filter(lambda x: x.keys() == {"fixed"}, fix_range["events"])
+                            )
+                            list_fixed = [i["fixed"] for i in filter_fixed]
+                            for i in list_fixed:
+                                try:
+                                    fixed_version.add(PypiVersion(i))
+                                except InvalidVersion:
+                                    logger.error(
+                                        f"fixed_version InvalidVersion - PypiVersion - {raw_data['id'] !r}"
+                                    )
+
+                        if fix_range["type"] == "SEMVER":
+                            filter_fixed = list(
+                                filter(lambda x: x.keys() == {"fixed"}, fix_range["events"])
+                            )
+                            list_fixed = [i["fixed"] for i in filter_fixed]
+
+                            for i in list_fixed:
+                                try:
+                                    fixed_version.add(SemverVersion(i))
+                                except InvalidVersion:
+                                    logger.error(
+                                        f"fixed_version InvalidVersion - SemverVersion - {raw_data['id'] !r}"
+                                    )
+
+                        # TODO add GitHubVersion univers fix_version
```

Comment:
Not sure what this TODO means... but it could be best to create an issue and link the issue instead?

Reply:
some affected package have as fixed_version as GIT . Does [univers](https://github.com/nexB/univers) have GitVersion ?
  ```
 "ranges": [
            {
                "type": "GIT",
                "repo": "https://github.com/unicode-org/icu.git",
                "events": [
                  { "introduced": "6e5755a2a833bc64852eae12967d0a54d7adf629" },
                  { "fixed": "c43455749b914feef56b178b256f29b3016146eb" }
                ]
            }
        ]
```

Reply:
Nope ... good catch!... we need to add this. This is going to be weird versioning scheme as you cannot determine if a commit is within a range, or before or after another commit using only the commit information.... You need a full git log!
As a first step, any such comparison can return a NotImplemented exception.
Later we can devise how to best handle these